pkg/client/typed/discovery/discovery_client.go

```diff
@@ -203,7 +229,10 @@ func (d *DiscoveryClient) serverPreferredResources(namespaced bool) ([]unversion
 			results = append(results, groupVersion.WithResource(apiResource.Name))
 		}
 	}
-	return results, utilerrors.NewAggregate(allErrs)
+	if len(failedGroups) > 0 {
+		return results, &ErrGroupDiscoveryFailed{Groups: failedGroups}
```

Comment:
Can you add a test case to cover this flow. While doing so please rename `client_test.go` to appropriate `discovery_client_test.go`.